backend/persistent/datastores/impala.js

```diff
@@ -0,0 +1,72 @@
+import fetch from 'node-fetch';
+import {createClient} from 'node-impala';
+import {dissoc, keys, values, init, map, prepend, unnest} from 'ramda';
+
+import Logger from '../../logger';
+
+const client = createClient();
+
+export function connect(connection) {
+    return client.connect({
+        host: connection.host,
+        port: connection.port,
+        resultType: 'json-array'
+    });
+}
+
+export function tables(connection) {
+    const code = (connection.database) ?
+        `show tables in ${connection.database}` :
+        'show tables';
+    return client.query(code)
+    .then(json => {
+        let tableNames = json.map(t => t.name);
+        if (connection.database) tableNames = tableNames.map(tn => `${connection.database}.${tn}`);
+        tableNames = tableNames.map(tn => tn.toUpperCase());
```

Comment:
why `toUpperCase`?

Reply:
It is a general convention to write SQL queries in uppercase, like:

`SELECT * FROM TABLE_NAME` -- so if tableNames are also capitalized, it fits right in the default queries. NBD though, since SQL is case-insensitive..

Reply:
👍